perf.go

```diff
@@ -283,33 +305,53 @@ func (pr *PerfReader) LostSamples() uint64 {
 	return atomic.LoadUint64(&pr.lostSamples)
 }
 
-// Close stops the reader.
+// Close stops the reader, discarding any samples not yet written to 'Samples'.
 //
 // Calls to perf_event_output from eBPF programs will return
 // ENOENT after calling this method.
 func (pr *PerfReader) Close() (err error) {
+	// Interrupt polling so we don't deadlock if the consumer is dead
+	close(pr.stopWriter)
```

Comment:
That has to be in a sync.Once block, otherwise this panics when calling close multiple times.

Reply:
Maybe make close() take a bool? `close(flush bool)` and then swap out the fds in close accordingly.

Not a fan of bool arguments, but here it's not exported and fairly benign.